crates/query-engine/translation/src/translation/query/root.rs

```diff
@@ -116,10 +147,159 @@ pub fn translate_rows_query(
             }
         })
         .collect::<Result<Vec<_>, Error>>()?;
+    Ok(columns)
+}
+
+/// Translate a nested field selection.
+///
+/// Nested fields are different from relationships in that the value of a nested field is already
+/// avaliable on the current table as a column of composite type.
+///
+/// A nested field selection translates to SQL in the form of:
+///
+///   SELECT
+///     coalesce(json_agg(row_to_json("%6_rows")), '[]') AS "rows"
+///   FROM
+///     (
+///       SELECT
+///         "%3_nested_fields_collect"."collected" AS "result"
+///       FROM
+///         <current table> AS "%0_<current table>"
+///         LEFT OUTER JOIN LATERAL (
+///           SELECT
+///             ("%0_<current table>"."<composite column>").*
+///         ) AS "%2_nested_field_bound" ON ('true')
+///         LEFT OUTER JOIN LATERAL (
+///           SELECT
+///             row_to_json("%4_nested_fields") AS "collected"
+///           FROM
+///             (
+///               SELECT
+///                 "%2_nested_field_bound"."<nested column>" AS "<nested field alias>"
+///             ) AS "%4_nested_fields"
+///         ) AS "%3_nested_fields_collect" ON ('true')
+///     ) AS "%6_rows"
+///
+fn translate_nested_field(
+    env: &Env,
+    state: &mut State,
+    // The table reference the column lives on
+    current_table: &TableNameAndReference,
+    // The column to extract nested fields from
```

Comment:
Can these be `///`? Does that work?

Reply:
It seems rust does not support `///` (or `#[doc()]` attributes) on function arguments. I've moved these to the main `///` block under an `# Aruguments` heading.

I'm fine either way myself.